### PR TITLE
An ML model Enhancement of Smartphone Price Prediction Model Using Dataset

### PR DESCRIPTION
**Pull request by BenakDeepak**
### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
**Please enter the title of the issue related to your pull request.**  
*Enter the issue title here.*
An ML model Enhancement of Smartphone Price Prediction Model Using Dataset
- [yes ] I have provided the issue title.

---

#### Info about the Related Issue
**What's the goal of the project?**  
*Describe the aim of the project.*
The aim of this enhancement is to improve the accuracy and performance of the existing machine learning model that predicts smartphone prices based on features such as ratings and the number of reviews. This includes refining data preprocessing, experimenting with different machine learning algorithms, and enhancing visualization for better insights.
- [yes ] I have described the aim of the project.

---

#### Name
**Please mention your name.**  
*Enter your name here.*
Benak Deepak
- [yes ] I have provided my name.

---

#### GitHub ID
**Please mention your GitHub ID.**  
*Enter your GitHub ID here.*
#151528559
- [ yes] I have provided my GitHub ID.

---

#### Email ID
**Please mention your email ID for further communication.**  
*Enter your email ID here.*
benakdeepak@gmail.com
- [x] I have provided my email ID.

---

#### Identify Yourself
**Mention in which program you are contributing (e.g., WoB, GSSOC, SSOC, SWOC).**  
*Enter your participant role here.*
GSSOC
- [ yes] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #issue-number*
#550
- [ yes] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
**Give a clear description of what you have added or modified.**  
*Describe your changes here.*

- [yes ] I have described my changes.
added a file in the machine learning folder
---

#### Type of Change
**Select the type of change:**  
- [ ] Bug fix (non-breaking change which fixes an issue)
- [yes ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

---

#### How Has This Been Tested?
**Describe how your changes have been tested.**  
*Describe your testing process here.*

- [ yes] I have described my testing process.
in localhost
---

#### Checklist
**Please confirm the following:**  
- [ yes] My code follows the guidelines of this project.
- [ yes] I have performed a self-review of my own code.
- [ yes] I have commented my code, particularly wherever it was hard to understand.
- [ yes] I have made corresponding changes to the documentation.
- [ yes] My changes generate no new warnings.
- [ yes] I have added things that prove my fix is effective or that my feature works.
- [ yes] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by BenakDeepak** [comment]
can u please review this

File: Machine_Learning/smartphone_rate_prediction/main.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import mean_squared_error, r2_score
+
+# Load the dataset
+df = pd.read_csv('C://Users//Deepak//Desktop//dataset.csv')  # Update with your actual file path
```
Comment (ruhi47) [inline Machine_Learning/smartphone_rate_prediction/main.py:10]: Hey, please provide an example dataset in the directory since it is crucial to test and run the program.
Will review the updated main.py and merge it.
Thank you!

**Review by ruhi47 — CHANGES_REQUESTED** [review]
